File: day_1/is_monotonic.py
testCase1 = []; #true

testCase2 = [1, 2, 3, 4]; #true

testCase3 = [3, 2, 1, 0]; #true

testCase4 = [1]; #true

testCase5 = [1, 2, 3, 2]; #false

testCase6 = [3, 2, 1, 4]; #false

testCase7 = [1, 3, 2, 4, 5] #false

def isMonotonicArr(arr):
  isMonotic = True
  isIncreasing = False
  isDecreasing = False

  if len(arr) == 0 or len(arr) == 1:
    return isMonotic
  
  if arr[1] > arr[0]:
    isIncreasing = True
  else: 
    isDecreasing = True
  
  for k in range(len(arr) - 1):
    current = arr[k]
    next = arr[k + 1]

    if (isIncreasing and current > next) or (isDecreasing and next > current):
      isMonotic = False
      return isMonotic

    
    
def monotonic_array(array):
    if len(array) == 0:
        return True
    first = array[0]
    last = array[len(array)-1]
    if first > last:
        for i in range(len(array)-1):
            if array[i] < array[i+1]:
                return False
    elif first == last:
        for i in range(len(array)-1):
            if array[i] != array[i+1]:
                return False
    else:
        for i in range(len(array)-1):
            if array[i] > array[i+1]:
                return False
    return True

    

print(1, monotonic_array(testCase1))

print(2, monotonic_array(testCase2))

print(3, monotonic_array(testCase3))

print(4, monotonic_array(testCase4))

print(5, monotonic_array(testCase5))

print(6, monotonic_array(testCase6))

print(6, monotonic_array(testCase7))

File: day_1/is_monotonic.js
const testCase1 = []; //true

const testCase2 = [1, 2, 3, 4]; //true

const testCase3 = [3, 2, 1, 0]; //true

const testCase4 = [1]; //true

const testCase5 = [1, 2, 3, 2]; //false

const testCase6 = [3, 2, 1, 4]; //false

const arrIsMonotonic = (arr) => {
  let isMonotonic = true,
    isIncreasing = false,
    isDecreasing = false;

  if ([0, 1].includes(arr.length)) return isMonotonic;

  const [first, second] = arr;

  if (first > second) {
    isDecreasing = true;
  } else {
    isIncreasing = true;
  }

  for (let i = 0; i < arr.length - 1; i++) {
    const current = arr[i];
    const next = arr[i + 1];

    if ((isIncreasing && current > next) || (isDecreasing && next > current)) {
      isMonotonic = false;
      break;
    }
  }

  return isMonotonic;
};

const checkMonotonic = function (arr) {
  //write code here to return either true or false

  if (arr.length <= 2) return true;

  let isIncreasing = true;

  let isDecreasing = true;

  for (var i = 1; i < arr.length - 1; i++) {
    if (arr[i] > arr[i - 1]) {
      isDecreasing = false;
    }

    if (arr[i] < arr[i - 1]) {
      isIncreasing = false;
    }
  }

  return isIncreasing || isDecreasing;
};

console.log(1, arrIsMonotonic(testCase1));

console.log(2, arrIsMonotonic(testCase2));

console.log(3, arrIsMonotonic(testCase3));

console.log(4, arrIsMonotonic(testCase4));

console.log(5, arrIsMonotonic(testCase5));

console.log(6, arrIsMonotonic(testCase6));
